streams/jvm/src/main/scala/zio/stream/experimental/platform.scala

```diff
@@ -119,4 +125,288 @@ trait ZStreamPlatformSpecificConstructors {
   ): ZStream[R, E, A] =
     effectAsyncInterrupt(k => register(k).toRight(UIO.unit), outputBuffer)
 
+  /**
+   * Creates a stream of bytes from a file at the specified path.
+   */
+  def fromFile(path: => Path, chunkSize: Int = ZStream.DefaultChunkSize): ZStream[Any, Throwable, Byte] =
+    ZStream
+      .bracket(ZIO.effectBlockingInterrupt(FileChannel.open(path)))(chan => ZIO.effectBlocking(chan.close()).orDie)
+      .flatMap { channel =>
+        ZStream.fromEffect(UIO(ByteBuffer.allocate(chunkSize))).flatMap { reusableBuffer =>
+          ZStream.repeatEffectChunkOption(
+            for {
+              bytesRead <- ZIO.effectBlockingInterrupt(channel.read(reusableBuffer)).asSomeError
+              _         <- ZIO.fail(None).when(bytesRead == -1)
+              chunk <- UIO {
+                         reusableBuffer.flip()
+                         Chunk.fromByteBuffer(reusableBuffer)
+                       }
+            } yield chunk
+          )
+        }
+      }
+
+  /**
+   * Creates a stream from a `java.io.InputStream`.
+   * Note: the input stream will not be explicitly closed after it is exhausted.
```

Comment:
What is the rationale for not closing InputStream?

Reply:
I think it is correct this way, closing the `InputStream` should be the responsibility of the creator (for example it could be coming from a `ZManaged`)